src/hotspot/share/gc/g1/heapRegion.inline.hpp

```diff
@@ -251,25 +221,51 @@ inline void HeapRegion::apply_to_marked_objects(G1CMBitMap* bitmap, ApplyToMarke
   assert(next_addr == limit, "Should stop the scan at the limit.");
 }
 
-inline HeapWord* HeapRegion::par_allocate_no_bot_updates(size_t min_word_size,
-                                                         size_t desired_word_size,
-                                                         size_t* actual_word_size) {
-  assert(is_young(), "we can only skip BOT updates on young regions");
+inline HeapWord* HeapRegion::par_allocate(size_t min_word_size,
+                                          size_t desired_word_size,
+                                          size_t* actual_word_size) {
   return par_allocate_impl(min_word_size, desired_word_size, actual_word_size);
 }
 
-inline HeapWord* HeapRegion::allocate_no_bot_updates(size_t word_size) {
+inline HeapWord* HeapRegion::allocate(size_t word_size) {
   size_t temp;
-  return allocate_no_bot_updates(word_size, word_size, &temp);
+  return allocate(word_size, word_size, &temp);
 }
 
-inline HeapWord* HeapRegion::allocate_no_bot_updates(size_t min_word_size,
-                                                     size_t desired_word_size,
-                                                     size_t* actual_word_size) {
-  assert(is_young(), "we can only skip BOT updates on young regions");
+inline HeapWord* HeapRegion::allocate(size_t min_word_size,
+                                      size_t desired_word_size,
+                                      size_t* actual_word_size) {
   return allocate_impl(min_word_size, desired_word_size, actual_word_size);
 }
 
+inline HeapWord* HeapRegion::bot_threshold_for_addr(const void* addr) {
+  assert(_bot_part.threshold_for_addr(addr) >= addr,
+         "threshold must be at or after given address. " PTR_FORMAT " >= " PTR_FORMAT,
+         p2i(_bot_part.threshold_for_addr(addr)), p2i(addr));
+  assert(is_old(),
+         "Should only calculate BOT threshold for old regions. addr: " PTR_FORMAT " region:" HR_FORMAT,
+         p2i(addr), HR_FORMAT_PARAMS(this));
+  return _bot_part.threshold_for_addr(addr);
```

Comment:
Instead of calling `_bot_part.threshold_for_addr` multiple times, storing the result in a local var could be cleaner.